api/services/deckService.js

```diff
@@ -0,0 +1,26 @@
+
+export default class Deck{
+    constructor(whiteCardDeck, players) {
+        this.whiteCardDeck = whiteCardDeck.map(cards => JSON.parse(JSON.stringify(cards)))
+        this.players = players;
+        this.shuffledDeck = this.shuffle(this.whiteCardDeck);
+        console.log(this.shuffledDeck);
+    }
+    dealCards(whiteCardDeck) {
+        for(let currentPlayer of this.players) {
+                for(let card of whiteCardDeck) {
+                    if (card.state === 'inDeck' && currentPlayer.whitecards.length < 10){
```

Comment:
it might be slightly nicer if you could have something like `(if whitecards.length == 10) break;` so it saves you iterating pointlessly